tollway/utils.py

```diff
@@ -1,23 +1,37 @@
-import random
 import json
+import random
 from datetime import datetime, timedelta
 from pathlib import Path
 
-from tollway.constants import TIMESTAMP_FORMAT
+from google.cloud import pubsub_v1
+
+from tollway.constants import (
+    DATE_VARIATION_MAX,
+    DATE_VARIATION_MIN,
+    PROJECT_ID,
+    TIMESTAMP_FORMAT,
+    TOPIC_ID,
+)
 
 
 def get_date_variation(timestamp: str) -> str:
-    random_integer = random.randint(1, 3)
+    random_integer = random.randint(DATE_VARIATION_MIN, DATE_VARIATION_MAX)
     current_timestamp = datetime.strptime(timestamp, TIMESTAMP_FORMAT)
     updated_timestamp = (current_timestamp - timedelta(days=random_integer)).strftime(TIMESTAMP_FORMAT)
     return updated_timestamp
 
 
-def push_to_topic():
-    pass # to be developed
+def get_topic():
+    publisher = pubsub_v1.PublisherClient()
+    topic_path = publisher.topic_path(project=PROJECT_ID, topic=TOPIC_ID)
+    return publisher, topic_path
+
+
+def encode_message(payload: dict) -> json:
```

Comment:
this type hint seems incorrect (are you running mypy?)

<img width="768" alt="image" src="https://github.com/BenGriffith/tollway-traffic/assets/387927/16fc204a-988d-4b07-9b5b-b7136f2d7869">

Reply:
awesome. Thanks. No, I'm not running it but I will work on adding/running it.

Reply:
wanted to add I really like typing/type hints but sometimes feel like it's hard to keep up with it all especially as the codebase grows. Implementing mypy I'm sure will help a lot.

Reply:
Yep + add it to pre-commit so it automatically runs.
Yeah type hints can add extra work and it's one common critique, on the other hand you can type hints as little or much as you want, it's all optional.

Reply:
gotcha, thank you. I just finished running mypy on all project files and working through errors. I like it, just a lot initially. I'll add to pre-commit also. I remember some of Will's code clinics on mypy and how in depth he went - blew my mind.

Reply:
Haha Will will like that feedback :)
We also have some good articles on the blog ...

Reply:
And a learning path on codechalleng.es in progress ...

Reply:
Awesome! I'll check em out.